### PR TITLE
export bandit types

### PR DESCRIPTION
**Pull request by aarsilv**
_Eppo Internal_:
🎟️ **Ticket:** [FF-2585 - Export common bandit logger types](https://linear.app/eppo/issue/FF-2584/node-sdk-export-common-bandit-logger-types)

It's useful for developers to have public access to the types used to define attributes, bandit attributes, and actions.

### CODE REVIEW

File: src/bandit-logger.ts [hunk 1]
```diff
@@ -17,5 +17,5 @@ export interface IBanditEvent {
 }
 
 export interface IBanditLogger {
-  logBanditAction(assignment: IBanditEvent): void;
+  logBanditAction(banditEvent: IBanditEvent): void;
```
Comment (aarsilv) [inline src/bandit-logger.ts:20]: Giving this a better argument name

File: src/client/eppo-client.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export interface IEppoClient {
   getStringAssignment(
     flagKey: string,
     subjectKey: string,
-    subjectAttributes: Record<string, AttributeType>,
+    subjectAttributes: Attributes,
```
Comment (aarsilv) [inline src/client/eppo-client.ts:63]: taking advantage of our `Attributes` type

File: src/index.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import ApiEndpoints from './api-endpoints';
-import { logger } from './application-logger';
+import { logger as applicationLogger } from './application-logger';
```
Comment (aarsilv) [inline src/index.ts:2]: plays nicer with unused variable IDE warnings to rename this here than when exporting 🤷‍♂️

File: src/index.ts [hunk 4]
```diff
@@ -72,4 +78,7 @@ export {
   VariationType,
   AttributeType,
   Attributes,
+  ContextAttributes,
+  BanditSubjectAttributes,
+  BanditActions,
```
Comment (aarsilv) [inline src/index.ts:83]: Used by bandits. Example signature:

```ts
getBanditAction(
  flagKey: string,
  subjectKey: string,
  subjectAttributes: BanditSubjectAttributes,
  actions: BanditActions,
  defaultValue: string,
): { variation: string; action: string | null };
```

See type definitions [here](https://github.com/Eppo-exp/js-client-sdk-common/blob/main/src/types.ts).

File: src/types.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 export type ValueType = string | number | boolean | JSON;
 export type AttributeType = string | number | boolean;
 export type ConditionValueType = AttributeType | AttributeType[];
-export type Attributes = { [key: string]: AttributeType };
+export type Attributes = Record<string, AttributeType>;
```
Comment (aarsilv) [inline src/types.ts:4]: Making this consistent with how we do this in other places.

**Review by felipecsl — APPROVED** [review]
I think this requires bumping the version to v3.6.0

**Pull request merged**